### PR TITLE
feature/SGRD-2_ability_to_test_jwt_auth_guard

### PR DESCRIPTION
**Pull request by amrography**
# Testing

You can generate JWT token for testing. It will be generated with [private_key](./tests/keys/private_key) from tests folder. And will be compared with `public_key` on same folder as well. **example**

```php
use Tests\TestCase;
use Supaapps\Guard\Tests\Concerns\GenerateJwtToken;

class CustomTest extends TestCase
{
    use GenerateJwtToken;

    public function testThatIAmActingAsUser(): void
    {
        $user = User::create();

        $this->withAccessTokenFor($user);

        $this->assertTrue(auth('jwt')->check());
        $this->assertTrue($user->id, auth('jwt')->id());
    }
}
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,46 +1,74 @@
...
 Route::middleware('auth:jwt')->get('/user', function (Request $request) {
     return [
         $request->user(),
-        auth()->firstName(),
-        auth()->lastName(),
-        auth()->email(),
-        auth()->scopes(),
-        auth()->scopesArray(),
-        ];
+        auth('jwt')->firstName(),
```
Comment (SirNarsh) [inline README.md]: with an assumption that by default APIs will have one driver `auth()` should be valid unless someone installs multiple drivers they know they have multiple drivers. or let's keep both mentioned

Comment (amrography) [reply README.md]: We can mention in docs to set the default `gaurd` to `jwt`

```php
'defaults' => [
        'guard' => 'jwt',
        'passwords' => 'users',
],
```

Comment (SirNarsh) [reply README.md]: let's add it as a comment, but assume if someone is installing package it's on new project as default, then add a note if it's not default how to use

File: README.md [hunk 2]
```diff
@@ -1,37 +1,50 @@
...
+- [Testing](#testing)
 
+## Installation
 
 `composer require supaapps/supaapps-guard`
 
+### ENV vars
 
-env vars:
+add env vars to your `.evn`:
```
Comment (SirNarsh) [inline README.md]: .env

File: README.md [hunk 3]
```diff
@@ -1,37 +1,50 @@
...
-        ],
+'guards' => [
+    'jwt' => [
+        'driver' => 'supaapps-guard',
+        'provider' => 'users',
     ],
+],
+```
 
+Also, set the default guard to `jwt`
```
Comment (SirNarsh) [inline README.md:35]: ~+ if not set as default, every-time you use `auth()` remember to pass driver name parameter: `auth('jwt')`~

File: README.md [hunk 4]
```diff
@@ -40,7 +53,32 @@ Route::middleware('auth:jwt')->get('/user', function (Request $request) {
         auth()->email(),
         auth()->scopes(),
         auth()->scopesArray(),
-        ];
+    ];
 });
+```
+
+*note*: `auth()` uses the default drive by default. If you didn't set the `jwt` as default driver then you need to call `auth('jwt')` on the previous usage example
```
Comment (SirNarsh) [inline README.md:60]: 👍

File: README.md (hunk 3 above)
Comment (SirNarsh) [reply README.md:35]: never mind saw you added that already below

**Review by SirNarsh — APPROVED** [review]

**Pull request merged**